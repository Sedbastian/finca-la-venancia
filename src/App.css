@font-face {
  font-family: Tangerine;
  src: url(./components/styles/Tangerine-Bold.ttf);
}

@font-face {
  font-family: Gloria;
  src: url(./components/styles/GloriaHallelujah-Regular.ttf);
}

@font-face {
  font-family: PlayFair;
  src: url(./components/styles/PlayfairDisplay-Italic-VariableFont_wght.ttf);
}

@font-face {
  font-family: Montserrat;
  src: url(./components/styles/Montserrat-Medium.ttf);
}

@font-face {
  font-family: Aoboshi;
  src: url(./components/styles/AoboshiOne-Regular.ttf);
}
@font-face {
  font-family: DancingScript;
  src: url(./components/styles/DancingScript-Regular.ttf);
}
div#root > div.appContainer {
  min-height: 100vh;

  display: flex;
  flex-direction: column;
  justify-content: space-between;
  gap: 1rem;

  background-image: url("./components/images/olivos/179.jpg");
  background-repeat: no-repeat;
  background-position: center;
  background-size: cover;
}
