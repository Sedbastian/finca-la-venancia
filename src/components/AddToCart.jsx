import "./styles/AddToCart.css";
import { useState, useContext } from "react";
import { CartDispatchContext } from "../CartContext";

export default function AddToCart({ inCart = false, name, quantityProp }) {
  const [quantity, setQuantity] = useState(quantityProp);
  const dispatch = useContext(CartDispatchContext);

  const [notAddedQ, setNotAddedQ] = useState(false);

  function addToCart(q) {
    dispatch({
      type: "added",
      name,
      quantity: q
    });
    setNotAddedQ(false);
  }

  function removeFromCart() {
    dispatch({
      type: "removed",
      name
    });
  }

  function decreaseQuantity() {
    if (inCart && quantity > 0) {
      addToCart(quantity - 1);
    }
    setQuantity(prevQuantity => {
      if (prevQuantity > 0) {
        return prevQuantity - 1;
      } else return 0;
    });
    if (!inCart && !notAddedQ) {
      setNotAddedQ(true);
    }
  }

  function incrementQuantity() {
    if (inCart) {
      addToCart(quantity + 1);
    }
    setQuantity(prevQuantity => {
      return prevQuantity + 1;
    });
    if (!inCart && !notAddedQ) {
      setNotAddedQ(true);
    }
  }

  return (
    <div className="addContainer">
      <div className="quantityContainer">
        <div className="quantityTitle">Cantidad</div>
        <button className="changeQuantity" onClick={decreaseQuantity}>
          â€“
        </button>
        <input
          type="number"
          name="quantity"
          value={`${quantity}`}
          onChange={e => setQuantity(e.target.value)}
        />
        <button className="changeQuantity" onClick={incrementQuantity}>
          +
        </button>
      </div>
      {inCart === false ? (
        <button
          className={!notAddedQ ? "addToCart" : "addToCart notAddedQ"}
          onClick={() => addToCart(quantity)}
        >
          Agregar al carrito
        </button>
      ) : (
        <button className="removeFromCart" onClick={removeFromCart}>
          Eliminar del carrito
        </button>
      )}
    </div>
  );
}
