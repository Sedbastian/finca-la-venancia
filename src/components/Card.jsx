import "./styles/Card.css";

export default function Card({ product, children }) {
  const { category, name, images, description, price, unit } = product;

  return (
    <li className="card">
      <div className="category">{category}</div>
      <div className="productName">{name}</div>
      <div className="imgCont">
        {images.map(image => {
          return <img key={image} src={image} alt="Producto" />;
        })}
      </div>
      <div className="description">{description}</div>
      <div className="price">
        ${price} cada {unit}
      </div>
      {children}
    </li>
  );
}
