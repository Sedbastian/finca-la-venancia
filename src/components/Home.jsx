import "./styles/Home.css";
import { Link } from "react-router-dom";
import { useEffect } from "react";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import {
  faBottleDroplet,
  faBrain,
  faJar
} from "@fortawesome/free-solid-svg-icons";
import foto1 from "./images/background/101_1215.JPG";
import foto2 from "./images/background/101_0518.JPG";
import foto3 from "./images/background/101_1221.JPG";

export default function Home() {
  useEffect(() => {
    window.scrollTo(0, 0);
  });

  return (
    <main className="home">
      <div className="homeContainer">
        <div className="pImgCont">
          <div className="imgContainer">
            <img src={foto3} alt="Olivos por Van Gogh" />
          </div>
          <p>
            {"    "}
            La Venancia esta ubicada en el pueblo de Santa Vera Cruz, al norte
            del departamento Castro Barros.
          </p>
        </div>
        <div className="pImgCont">
          <p>
            {"    "}
            Compuesta de un hermoso y viejo olivar de 160 olivos en convivencia
            con una gran diversidad de otros frutales como nogales, vid,
            duraznos, damascos, membrillos, higueras, todo conviviendo en
            armonía con el monte nativo.
          </p>
          <div className="imgContainer">
            <img src={foto1} alt="Olivos por Van Gogh" />
          </div>
        </div>
        <div className="pImgCont">
          <div className="imgContainer">
            <img src={foto2} alt="Olivos por Van Gogh" />
          </div>
          <p>
            {"    "}
            Hay también muchas aromaticas y tenemos espacios para huerta.
          </p>
        </div>
      </div>
      <Link to={`catalog`} className="catalogButton">
        <FontAwesomeIcon icon={faBottleDroplet} />
        <FontAwesomeIcon icon={faBrain} />
        <FontAwesomeIcon icon={faJar} /> Catálogo de Productos
      </Link>
    </main>
  );
}
