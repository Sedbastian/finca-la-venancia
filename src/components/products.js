import nueces from "./images/products/nueces.jpg";
import aceite from "./images/products/nueces.jpg";

const products = new Map();

products.set("Nueces Peladas", {
  category: "Productos Principales",
  name: "Nueces Peladas",
  image: nueces,
  description: "Agroecológicas, criollas, mariposa",
  price: 2000,
  quantity: 1,
  unit: "kilo"
});

products.set("Aceite de Oliva Virgen Extra", {
  category: "Productos Principales",
  name: "Aceite de Oliva Virgen Extra",
  image: aceite,
  description: "Variedades: Arauco, Arbequina, Manzanilla",
  quantity: 1,
  price: 2000,
  unit: "litro"
});

products.set("Jalea de Membrillo", {
  category: "Dulces",
  name: "Jalea de Membrillo",
  image: aceite,
  description: "Ingredientes: Membrillos Agroecológicos, azucar, jugo de limón",
  quantity: 1,
  price: 1000,
  unit: "frasco de 330cc"
});

export default products;
