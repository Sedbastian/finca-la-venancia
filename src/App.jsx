import "./App.css";
import { Outlet } from "react-router-dom";
import { useReducer } from "react";
import { CartContext, CartDispatchContext } from "./CartContext";
import Header from "./components/Header";
import Footer from "./components/Footer";

function App() {
  const [cart, dispatch] = useReducer(cartReducer, new Map());

  function cartReducer(cart, action) {
    switch (action.type) {
      case "added": {
        const updatedCart = new Map(
          cart.set(action.name, {
            name: action.name,
            quantity: action.quantity
          })
        );
        return updatedCart;
      }
      case "removed": {
        cart.delete(action.name);
        const updatedCart = new Map(cart);
        return updatedCart;
      }
      default:
        throw Error("Unknown action: " + action.type);
    }
  }

  return (
    <div className="appContainer">
      <CartContext.Provider value={cart}>
        <CartDispatchContext.Provider value={dispatch}>
          <Header />
          <Outlet />
          <Footer />
        </CartDispatchContext.Provider>
      </CartContext.Provider>
    </div>
  );
}

export default App;
