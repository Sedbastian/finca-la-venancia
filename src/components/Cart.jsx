import "./styles/Cart.css";
import { useContext } from "react";
import { CartContext } from "../CartContext";
import products from "./products";
import Card from "./Card";
import AddToCart from "./AddToCart";

export default function Cart() {
  const cart = useContext(CartContext);

  let totalAmount = 0;
  const billItems = [...cart.values()].map(item => {
    // Add item quantity to the item in products Map
    const product = products.get(item.name);
    products.set(item.name, {
      ...product,
      quantity: item.quantity
    });
    totalAmount += product.quantity * product.price;

    return (
      <li key={`billItem-${product.name}`}>
        <div className="number">{product.quantity}</div>
        <div>{product.name}</div>
        <div className="number">${product.price}</div>
        <div>{product.unit}</div>
        <div className="number productMult">${product.quantity * product.price}</div>
      </li>
    );
  });

  const totalAmountLi = (
    <li className="totalAmount">
      <div></div>
      <div>TOTAL</div>
      <div></div>
      <div></div>
      <div className="number total">${totalAmount}</div>
    </li>
  );

  const cartCards = [...cart.values()].map(item => {
    return (
      <Card key={item.name} product={products.get(item.name)}>
        <AddToCart
          inCart={true}
          name={item.name}
          quantityProp={item.quantity}
        />
      </Card>
    );
  });

  return (
    <main className="cart">
      <h2>Carrito de Compras</h2>
      {cart.size === 0 ? (
        <p>El carrito est√° vacio...</p>
      ) : (
        <ul className="cartTable">
          <li className="head">
            <div className="number">Cantidad</div>
            <div>Producto</div>
            <div className="number">Precio por unidad</div>
            <div>Unidad</div>
            <div className="number productMultHead">Subtotales</div>
          </li>
          {billItems}
          {totalAmountLi}
        </ul>
      )}
      <ul className="cartList">{cartCards}</ul>
    </main>
  );
}
