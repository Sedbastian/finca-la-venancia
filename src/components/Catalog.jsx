import { useEffect } from "react";
import products from "./products";
import Card from "./Card";
import AddToCart from "./AddToCart";

export default function Catalog() {
  useEffect(() => {
    window.scrollTo(0, 0);
  });

  return (
    <main className="catalog">
      <ul className="productsList">
        {[...products.values()].map(product => {
          return (
            <Card key={product.name} product={product}>
              <AddToCart name={product.name} quantityProp={product.quantity} />
            </Card>
          );
        })}
      </ul>
    </main>
  );
}
