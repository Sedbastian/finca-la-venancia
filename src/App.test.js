import { createHashRouter, RouterProvider } from "react-router-dom";
import React from "react";

import "@testing-library/jest-dom";
import "@testing-library/jest-dom/extend-expect";
import userEvent from "@testing-library/user-event";
import { render, screen } from "@testing-library/react";

// Components
import App from "./App";
import ErrorPage from "./components/ErrorPage";
import Home from "./components/Home";
import Catalog from "./components/Catalog";
import Cart from "./components/Cart";
import Card from "./components/Card";
import AddToCart from "./components/AddToCart";
import products from "./components/products";

// Mocks
window.scrollTo = jest.fn();

jest.mock("./components/products", () => {
  const productsMock = new Map();

  productsMock.set("Nueces Peladas", {
    category: "Productos Principales",
    name: "Nueces Peladas",
    images: [],
    description: "Agroecológicas, criollas, mariposa",
    price: 2000,
    quantity: 1,
    unit: "kilo"
  });

  productsMock.set("Aceite de Oliva Virgen Extra", {
    category: "Productos Principales",
    name: "Aceite de Oliva Virgen Extra",
    images: [],
    description: "Variedades: Arauco, Arbequina, Manzanilla",
    quantity: 1,
    price: 2000,
    unit: "litro"
  });

  productsMock.set("Aceitunas Negras en Salmuera", {
    category: "Productos Principales",
    name: "Aceitunas Negras en Salmuera",
    images: [],
    description:
      "Aceitunas variedad Arauco, maduras, negras, agroecológicas en salmuera al 6%",
    quantity: 1,
    price: 1000,
    unit: "frasco de 1 kilo escurrido"
  });

  return productsMock;
});

describe("App component", () => {
  it("Matches snapshot", () => {
    const router = createHashRouter([
      {
        path: "/",
        element: <App />,
        errorElement: <ErrorPage />,
        children: [
          { index: true, element: <Home /> },
          {
            path: "catalog",
            element: <Catalog />
          },
          {
            path: "cart",
            element: <Cart />
          }
        ]
      }
    ]);

    const { container } = render(<RouterProvider router={router} />);
    expect(container).toMatchSnapshot();
  });
});

describe("Header component", () => {
  it("renders correct heading", () => {
    const router = createHashRouter([
      {
        path: "/",
        element: <App />,
        errorElement: <ErrorPage />,
        children: [
          { index: true, element: <Home /> },
          {
            path: "catalog",
            element: <Catalog />
          },
          {
            path: "cart",
            element: <Cart />
          }
        ]
      }
    ]);

    render(<RouterProvider router={router} />);

    expect(screen.getByRole("heading").textContent).toMatch(
      /finca agroecológica “la venancia”/i
    );
  });

  it("Links change location correctly", async () => {
    const router = createHashRouter([
      {
        path: "/",
        element: <App />,
        errorElement: <ErrorPage />,
        children: [
          { index: true, element: <Home /> },
          {
            path: "catalog",
            element: <Catalog />
          },
          {
            path: "cart",
            element: <Cart />
          }
        ]
      }
    ]);

    const user = userEvent.setup();

    render(<RouterProvider router={router} />);

    const headerCatalogLink = screen.getByRole("link", {
      name: "Catálogo"
    });
    await user.click(headerCatalogLink);
    expect(document.location.hash).toBe("#/catalog");

    const headerCartLink = screen.getByRole("link", { name: "Carrito" });
    await user.click(headerCartLink);
    expect(document.location.hash).toBe("#/cart");

    const headerHomeLink = screen.getByRole("link", { name: "Inicio" });
    await user.click(headerHomeLink);
    expect(document.location.hash).toBe("#/");
  });
});

describe("Home component", () => {
  it("Catalog Link changes location correctly and Home Link on Header returns to '#/'", async () => {
    const router = createHashRouter([
      {
        path: "/",
        element: <App />,
        errorElement: <ErrorPage />,
        children: [
          { index: true, element: <Home /> },
          {
            path: "catalog",
            element: <Catalog />
          },
          {
            path: "cart",
            element: <Cart />
          }
        ]
      }
    ]);

    const user = userEvent.setup();

    render(<RouterProvider router={router} />);

    const homeCatalogLink = screen.getByRole("link", {
      name: "Catálogo de Productos"
    });
    await user.click(homeCatalogLink);
    expect(document.location.hash).toBe("#/catalog");

    const homeLink = screen.getByRole("link", { name: "Inicio" });
    await user.click(homeLink);
    expect(document.location.hash).toBe("#/");
  });
});

describe("Footer component", () => {
  it("Links point to correct addresses", async () => {
    const router = createHashRouter([
      {
        path: "/",
        element: <App />,
        errorElement: <ErrorPage />,
        children: [
          { index: true, element: <Home /> },
          {
            path: "catalog",
            element: <Catalog />
          },
          {
            path: "cart",
            element: <Cart />
          }
        ]
      }
    ]);

    const user = userEvent.setup();

    render(<RouterProvider router={router} />);

    const sedbastianLink = screen.getByRole("link", {
      name: "Sedbastian"
    });
    expect(sedbastianLink.href).toBe("https://github.com/Sedbastian");

    const repoLink = screen.getByRole("link", { name: "githubRepo" });
    expect(repoLink.href).toBe("https://github.com/Sedbastian/shopping-cart");
  });
});

describe("Catalog component", () => {
  it("Shows correct number of products", () => {
    render(<Catalog />);

    const productsCards = screen.getAllByRole("listitem");

    expect(productsCards.length).toBe(products.size);

    productsCards.forEach(card => {
      const productName = card.querySelector(".productName");
      expect(products.has(productName.textContent)).toBe(true);
    });
  });
});

describe("AddToCart component", () => {
  it("Correctly displays input quantity when change quantity buttons are pressed", async () => {
    const user = userEvent.setup();

    render(<AddToCart name="Test product" quantityProp={8} />);

    const decrementQuantityButton = screen.getByRole("button", { name: "–" });
    const incrementQuantityButton = screen.getByRole("button", { name: "+" });
    const inputQuantity = screen.getByRole("spinbutton");

    await user.click(decrementQuantityButton);

    await user.click(incrementQuantityButton);
    await user.click(incrementQuantityButton);
    await user.click(incrementQuantityButton);
    await user.click(incrementQuantityButton);

    expect(inputQuantity.value).toBe("11");
  });

  it("Correctly adds className='notAddedQ' to AddToCart button when change quantity buttons are pressed", async () => {
    const user = userEvent.setup();

    render(<AddToCart name="Test product" quantityProp={8} />);

    const incrementQuantityButton = screen.getByRole("button", { name: "+" });
    const addToCartButton = screen.getByRole("button", {
      name: "Agregar al carrito"
    });

    await user.click(incrementQuantityButton);

    expect(addToCartButton.classList["1"]).toBe("notAddedQ");
  });
});

describe("System test", () => {
  it("Correctly adds products and show total amount", async () => {
    const router = createHashRouter([
      {
        path: "/",
        element: <App />,
        errorElement: <ErrorPage />,
        children: [
          {
            index: true,
            element: <Home />
          },
          {
            path: "catalog",
            element: <Catalog />
          },
          {
            path: "cart",
            element: <Cart />
          }
        ]
      }
    ]);

    const user = userEvent.setup();
    render(<RouterProvider router={router} />);

    // Go Home
    const homeLink = screen.getByRole("link", {
      name: "Inicio"
    });
    await user.click(homeLink);
    expect(document.location.hash).toBe("#/");

    // Go to Catalog
    const catalogButton = screen.getByRole("link", {
      name: "Catálogo de Productos"
    });
    await user.click(catalogButton);
    expect(document.location.hash).toBe("#/catalog");

    // Add to Cart 4 units of each product displayed
    const catalogList = screen.getAllByRole("listitem");
    catalogList.forEach(async card => {
      const addToCartButton = card.querySelector(".addToCart");
      await user.click(addToCartButton);
    });

    // Go to Cart
    const cartButton = screen.getByRole("link", {
      name: "Carrito"
    });
    await user.click(cartButton);
    expect(document.location.hash).toBe("#/cart");

    // cartLink shows correct number of added products
    expect(screen.getByTestId("cartQuantity").textContent).toBe("3");

    // cartTable shows correct totalAmomunt
    expect(screen.getByRole("generic", { name: "total" }).textContent).toBe(
      "$5000"
    );
  });
});
