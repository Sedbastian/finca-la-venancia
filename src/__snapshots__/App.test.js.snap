// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component Matches snapshot 1`] = `
<div>
  <div
    class="appContainer"
  >
    <header>
      <div
        class="container"
      >
        <h1>
          <div
            class="finca"
          >
            Finca Agroecológica
          </div>
           
          <div
            class="venancia"
          >
            “La Venancia”
          </div>
        </h1>
        <nav>
          <a
            class="homeLink currentTab"
            href="#/"
            title="Inicio"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-house-chimney "
              data-icon="house-chimney"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 576 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M543.8 287.6c17 0 32-14 32-32.1c1-9-3-17-11-24L512 185V64c0-17.7-14.3-32-32-32H448c-17.7 0-32 14.3-32 32v36.7L309.5 7c-6-5-14-7-21-7s-15 1-22 8L10 231.5c-7 7-10 15-10 24c0 18 14 32.1 32 32.1h32v69.7c-.1 .9-.1 1.8-.1 2.8V472c0 22.1 17.9 40 40 40h16c1.2 0 2.4-.1 3.6-.2c1.5 .1 3 .2 4.5 .2H160h24c22.1 0 40-17.9 40-40V448 384c0-17.7 14.3-32 32-32h64c17.7 0 32 14.3 32 32v64 24c0 22.1 17.9 40 40 40h24 32.5c1.4 0 2.8 0 4.2-.1c1.1 .1 2.2 .1 3.3 .1h16c22.1 0 40-17.9 40-40V455.8c.3-2.6 .5-5.3 .5-8.1l-.7-160.2h32z"
                fill="currentColor"
              />
            </svg>
          </a>
          <a
            class="catalogLink"
            href="#/catalog"
            title="Catálogo"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-bottle-droplet "
              data-icon="bottle-droplet"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 320 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M96 0C82.7 0 72 10.7 72 24s10.7 24 24 24c4.4 0 8 3.6 8 8v64.9c0 12.2-7.2 23.1-17.2 30.1C53.7 174.1 32 212.5 32 256V448c0 35.3 28.7 64 64 64H224c35.3 0 64-28.7 64-64V256c0-43.5-21.7-81.9-54.8-105c-10-7-17.2-17.9-17.2-30.1V56c0-4.4 3.6-8 8-8c13.3 0 24-10.7 24-24s-10.7-24-24-24l-8 0 0 0 0 0H104l0 0 0 0L96 0zm64 382c-26.5 0-48-20.1-48-45c0-16.8 22.1-48.1 36.3-66.4c6-7.8 17.5-7.8 23.5 0C185.9 288.9 208 320.2 208 337c0 24.9-21.5 45-48 45z"
                fill="currentColor"
              />
            </svg>
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-brain "
              data-icon="brain"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M184 0c30.9 0 56 25.1 56 56V456c0 30.9-25.1 56-56 56c-28.9 0-52.7-21.9-55.7-50.1c-5.2 1.4-10.7 2.1-16.3 2.1c-35.3 0-64-28.7-64-64c0-7.4 1.3-14.6 3.6-21.2C21.4 367.4 0 338.2 0 304c0-31.9 18.7-59.5 45.8-72.3C37.1 220.8 32 207 32 192c0-30.7 21.6-56.3 50.4-62.6C80.8 123.9 80 118 80 112c0-29.9 20.6-55.1 48.3-62.1C131.3 21.9 155.1 0 184 0zM328 0c28.9 0 52.6 21.9 55.7 49.9c27.8 7 48.3 32.1 48.3 62.1c0 6-.8 11.9-2.4 17.4c28.8 6.2 50.4 31.9 50.4 62.6c0 15-5.1 28.8-13.8 39.7C493.3 244.5 512 272.1 512 304c0 34.2-21.4 63.4-51.6 74.8c2.3 6.6 3.6 13.8 3.6 21.2c0 35.3-28.7 64-64 64c-5.6 0-11.1-.7-16.3-2.1c-3 28.2-26.8 50.1-55.7 50.1c-30.9 0-56-25.1-56-56V56c0-30.9 25.1-56 56-56z"
                fill="currentColor"
              />
            </svg>
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-jar "
              data-icon="jar"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 320 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M32 32C32 14.3 46.3 0 64 0H256c17.7 0 32 14.3 32 32s-14.3 32-32 32H64C46.3 64 32 49.7 32 32zM0 160c0-35.3 28.7-64 64-64H256c35.3 0 64 28.7 64 64V448c0 35.3-28.7 64-64 64H64c-35.3 0-64-28.7-64-64V160zm96 64c-17.7 0-32 14.3-32 32v96c0 17.7 14.3 32 32 32H224c17.7 0 32-14.3 32-32V256c0-17.7-14.3-32-32-32H96z"
                fill="currentColor"
              />
            </svg>
          </a>
          <a
            class="cartLink"
            href="#/cart"
            title="Carrito"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-basket-shopping "
              data-icon="basket-shopping"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 576 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M253.3 35.1c6.1-11.8 1.5-26.3-10.2-32.4s-26.3-1.5-32.4 10.2L117.6 192H32c-17.7 0-32 14.3-32 32s14.3 32 32 32L83.9 463.5C91 492 116.6 512 146 512H430c29.4 0 55-20 62.1-48.5L544 256c17.7 0 32-14.3 32-32s-14.3-32-32-32H458.4L365.3 12.9C359.2 1.2 344.7-3.4 332.9 2.7s-16.3 20.6-10.2 32.4L404.3 192H171.7L253.3 35.1zM192 304v96c0 8.8-7.2 16-16 16s-16-7.2-16-16V304c0-8.8 7.2-16 16-16s16 7.2 16 16zm96-16c8.8 0 16 7.2 16 16v96c0 8.8-7.2 16-16 16s-16-7.2-16-16V304c0-8.8 7.2-16 16-16zm128 16v96c0 8.8-7.2 16-16 16s-16-7.2-16-16V304c0-8.8 7.2-16 16-16s16 7.2 16 16z"
                fill="currentColor"
              />
            </svg>
          </a>
        </nav>
      </div>
    </header>
    <main
      class="home"
    >
      <div
        class="homeContainer"
      >
        <div
          class="pImgCont"
        >
          <div
            class="imgContainer"
          >
            <img
              alt="Olivos por Van Gogh"
              src="101_1221.JPG"
            />
          </div>
          <p>
                
            La Venancia esta ubicada en el pueblo de Santa Vera Cruz, al norte del departamento Castro Barros.
          </p>
        </div>
        <div
          class="pImgCont"
        >
          <p>
                
            Compuesta de un hermoso y viejo olivar de 160 olivos en convivencia con una gran diversidad de otros frutales como nogales, vid, duraznos, damascos, membrillos, higueras, todo conviviendo en armonía con el monte nativo.
          </p>
          <div
            class="imgContainer"
          >
            <img
              alt="Olivos por Van Gogh"
              src="101_1215.JPG"
            />
          </div>
        </div>
        <div
          class="pImgCont"
        >
          <div
            class="imgContainer"
          >
            <img
              alt="Olivos por Van Gogh"
              src="101_0518.JPG"
            />
          </div>
          <p>
                
            Hay también muchas aromaticas y tenemos espacios para huerta.
          </p>
        </div>
      </div>
      <a
        class="catalogButton"
        href="#/catalog"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-bottle-droplet "
          data-icon="bottle-droplet"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 320 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M96 0C82.7 0 72 10.7 72 24s10.7 24 24 24c4.4 0 8 3.6 8 8v64.9c0 12.2-7.2 23.1-17.2 30.1C53.7 174.1 32 212.5 32 256V448c0 35.3 28.7 64 64 64H224c35.3 0 64-28.7 64-64V256c0-43.5-21.7-81.9-54.8-105c-10-7-17.2-17.9-17.2-30.1V56c0-4.4 3.6-8 8-8c13.3 0 24-10.7 24-24s-10.7-24-24-24l-8 0 0 0 0 0H104l0 0 0 0L96 0zm64 382c-26.5 0-48-20.1-48-45c0-16.8 22.1-48.1 36.3-66.4c6-7.8 17.5-7.8 23.5 0C185.9 288.9 208 320.2 208 337c0 24.9-21.5 45-48 45z"
            fill="currentColor"
          />
        </svg>
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-brain "
          data-icon="brain"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M184 0c30.9 0 56 25.1 56 56V456c0 30.9-25.1 56-56 56c-28.9 0-52.7-21.9-55.7-50.1c-5.2 1.4-10.7 2.1-16.3 2.1c-35.3 0-64-28.7-64-64c0-7.4 1.3-14.6 3.6-21.2C21.4 367.4 0 338.2 0 304c0-31.9 18.7-59.5 45.8-72.3C37.1 220.8 32 207 32 192c0-30.7 21.6-56.3 50.4-62.6C80.8 123.9 80 118 80 112c0-29.9 20.6-55.1 48.3-62.1C131.3 21.9 155.1 0 184 0zM328 0c28.9 0 52.6 21.9 55.7 49.9c27.8 7 48.3 32.1 48.3 62.1c0 6-.8 11.9-2.4 17.4c28.8 6.2 50.4 31.9 50.4 62.6c0 15-5.1 28.8-13.8 39.7C493.3 244.5 512 272.1 512 304c0 34.2-21.4 63.4-51.6 74.8c2.3 6.6 3.6 13.8 3.6 21.2c0 35.3-28.7 64-64 64c-5.6 0-11.1-.7-16.3-2.1c-3 28.2-26.8 50.1-55.7 50.1c-30.9 0-56-25.1-56-56V56c0-30.9 25.1-56 56-56z"
            fill="currentColor"
          />
        </svg>
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-jar "
          data-icon="jar"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 320 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M32 32C32 14.3 46.3 0 64 0H256c17.7 0 32 14.3 32 32s-14.3 32-32 32H64C46.3 64 32 49.7 32 32zM0 160c0-35.3 28.7-64 64-64H256c35.3 0 64 28.7 64 64V448c0 35.3-28.7 64-64 64H64c-35.3 0-64-28.7-64-64V160zm96 64c-17.7 0-32 14.3-32 32v96c0 17.7 14.3 32 32 32H224c17.7 0 32-14.3 32-32V256c0-17.7-14.3-32-32-32H96z"
            fill="currentColor"
          />
        </svg>
         Catálogo de Productos
      </a>
    </main>
    <footer>
      <a
        class="sedbastian"
        href="https://github.com/Sedbastian"
      >
        Sedbastian
      </a>
      <a
        class="github"
        href="https://github.com/Sedbastian/shopping-cart"
        title="githubRepo"
      >
        <svg
          viewBox="0 0 496 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"
            fill="currentColor"
          />
        </svg>
      </a>
      <div
        class="year"
      >
        2023
      </div>
    </footer>
  </div>
</div>
`;
